src/App.tsx

```diff
@@ -15,36 +16,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortField {
+  Name = 'name',
+  Length = 'length',
+  NoField = '',
+}
+
+type SortProps = {
+  sortField: SortField;
+  reversed: boolean;
+};
+
+function getVisibleGoods(goods: string[], { sortField, reversed }: SortProps) {
+  let result = [...goods];
+
+  if (sortField) {
+    switch (sortField) {
+      case SortField.Name:
+        result = result.sort();
+        break;
+      case SortField.Length:
+        result = result.sort((good1, good2) => good1.length - good2.length);
+        break;
+      default:
+        break;
+    }
+  }
+
+  if (reversed) {
+    result.reverse();
+  }
+
+  return result;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState(SortField.NoField);
+  const [reversed, setReversed] = useState(false);
+
+  const visibleGoods = getVisibleGoods(goodsFromServer, {
+    sortField,
+    reversed,
+  });
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={`button is-info ${cn({ 'is-light': sortField !== SortField.Name })}`}
```

Comment:
You are somewhat misusing classnames:

```suggestion
          className={cn('button', 'is-info', { 'is-light': sortField !== SortField.Name })}
```

Please fix other cases too